src/QMCHamiltonians/tests/test_stress.cpp

```diff
@@ -0,0 +1,133 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Yubo "Paul" Yang, yyang173@illinois.edu, University of Illinois at Urbana-Champaign
+//
+// File created by: Yubo "Paul" Yang, yyang173@illinois.edu, University of Illinois at Urbana-Champaign
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#include "catch.hpp"
+
+#include "OhmmsData/Libxml2Doc.h"
+#include "OhmmsPETE/OhmmsMatrix.h"
+#include "Particle/ParticleSet.h"
+#include "Particle/ParticleSetPool.h"
+#include "LongRange/EwaldHandler3D.h"
+#include "QMCWaveFunctions/TrialWaveFunction.h"
+#include "QMCHamiltonians/StressPBC.h"
+
+#include <stdio.h>
+#include <string>
+
+using std::string;
+
+namespace qmcplusplus
+{
+// PBC case
+TEST_CASE("Stress BCC H Ewald3D", "[hamiltonian]")
+{
+  Communicate* c;
+  c = OHMMS::Controller;
+
+  CrystalLattice<OHMMS_PRECISION, OHMMS_DIM> Lattice;
+  Lattice.BoxBConds = true; // periodic
+  Lattice.R.diagonal(3.24957306);
+  Lattice.LR_dim_cutoff = 40;
+  Lattice.reset();
+
+
+  ParticleSet ions;
+  ParticleSet elec;
+
+  ions.setName("ion");
+  ions.create(2);
+  ions.R[0][0] = 0.0;
+  ions.R[0][1] = 0.0;
+  ions.R[0][2] = 0.0;
+  ions.R[1][0] = 1.62478653;
+  ions.R[1][1] = 1.62478653;
+  ions.R[1][2] = 1.62478653;
+
+
+  SpeciesSet& ion_species       = ions.getSpeciesSet();
+  int pIdx                      = ion_species.addSpecies("H");
+  int pChargeIdx                = ion_species.addAttribute("charge");
+  ion_species(pChargeIdx, pIdx) = 1;
+  ions.Lattice = Lattice;
+  ions.createSK();
+
+
+  elec.Lattice = Lattice;
+  elec.setName("elec");
+  elec.create(2);
+  elec.R[0][0] = 0.4;
+  elec.R[0][1] = 0.4;
+  elec.R[0][2] = 0.4;
+  elec.R[1][0] = 2.02478653;
+  elec.R[1][1] = 2.02478653;
+  elec.R[1][2] = 2.02478653;
+
+
+  SpeciesSet& tspecies         = elec.getSpeciesSet();
+  int upIdx                    = tspecies.addSpecies("u");
+  int chargeIdx                = tspecies.addAttribute("charge");
+  int massIdx                  = tspecies.addAttribute("mass");
+  tspecies(chargeIdx, upIdx)   = -1;
+  tspecies(massIdx, upIdx)     = 1.0;
+
+  elec.createSK();
+
+  ParticleSetPool ptcl = ParticleSetPool(c);
+
+  ions.resetGroups();
+
+  // The call to resetGroups is needed transfer the SpeciesSet
+  // settings to the ParticleSet
+  elec.resetGroups();
+
+  TrialWaveFunction psi(c);
+
+  LRCoulombSingleton::CoulombHandler = new EwaldHandler3D(ions);
```

Comment:
Use unique_ptr

Reply:
I am not familiar with unique_ptr. I tried `std::unique_ptr<LRCoulombSingleton::CoulombHandler>` and `dynamic_cast` the RHS. Neither compiled. Looking around the code base, I do not see an example for using unique_ptr with `new`. Could you point me to an example please?

Reply:
I refactored some existing code in #2593. The change pattern should be quite the same.

Reply:
@ye-luo Thanks a lot for #2593 . I updated this PR to use unique_ptr.